### ISSUE TITLE
[BITV]: adapt checkboxes colors in normal state on dark theme

### ISSUE THREAD

**Issue by JuliaKirschenheuter** [nextcloud/server#42121]
**Waiting for https://github.com/nextcloud/server/pull/42287 to be merged and backported**

![Screenshot from 2023-12-08 10-58-04](https://github.com/nextcloud/server/assets/6078378/9a7fe9d6-4bac-4e25-878c-596cd68b293a)

This have to be done for dark theme as well.

Problems: 

- Checkboxes
- Radiobuttons
- Checkbox radio switch

![Screenshot from 2023-12-08 13-36-03](https://github.com/nextcloud/server/assets/6078378/4aca0870-43eb-4d6b-bcb8-7d83330f2cc3)
![Screenshot from 2023-12-08 13-35-41](https://github.com/nextcloud/server/assets/6078378/867cc794-27b9-42d2-a8ac-45d757d11d26)
![Screenshot from 2023-12-08 13-47-30](https://github.com/nextcloud/server/assets/6078378/10a52b12-483e-421e-91ad-57be56737337)
![Screenshot from 2023-12-08 13-48-27](https://github.com/nextcloud/server/assets/6078378/1f3cee89-04be-452d-99ac-54b5c33236ea)

Suggestion: 

Use `--color-primary-element-light-text` instead of `--color-primary`, like:

![Screenshot from 2023-12-08 11-58-46](https://github.com/nextcloud/server/assets/6078378/cd202f4a-540f-4095-bc2e-876d10a8e1ce)

**Comment by szaimen** [comment]
I guess the calculation of the colors indeed needs to be different on dark mode compared to light mode in order to fullfill the contrast requirements. Calculation is done in https://github.com/nextcloud/server/blob/da97e10b658b4603e5847f31739c66eefef8d862/apps/theming/lib/Themes/CommonThemeTrait.php#L41-L82 IIRC...

**Comment by JuliaKirschenheuter** [comment]
Thank you @szaimen!

Do we really need to adjust this colors? It looks to me that i can just use an existing one `--color-primary-element-light-text` like if is used for text inside of the button: 
![Screenshot from 2023-12-08 15-52-16](https://github.com/nextcloud/server/assets/6078378/718525e7-2651-4248-91ff-5b0a5c6433a3)

and then use a dark mode mixin. What do you think?

**Comment by szaimen** [comment]
> Do we really need to adjust this colors? It looks to me that i can just use an existing one `--color-primary-element-light-text` like if is used for text inside of the button: ![Screenshot from 2023-12-08 15-52-16](https://private-user-images.githubusercontent.com/6078378/289105468-718525e7-2651-4248-91ff-5b0a5c6433a3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIwNDgwNTMsIm5iZiI6MTcwMjA0Nzc1MywicGF0aCI6Ii82MDc4Mzc4LzI4OTEwNTQ2OC03MTg1MjVlNy0yNjUxLTQyNDgtOTFmZi01YjBhNWM2NDMzYTMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIwOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMDhUMTUwMjMzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NmZkMDI1MzUwOTZmMGY0Yzk0NDJhOTg3YjMxNWYyZTUxMTU5MTY1NzIwMzlkZDFjMTBhNDUwOTZkYWE0Mzk3MiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.bEx6pM47LnLcWD2J8Zm8VPhDF96Gk-lDyqqxi_DMokA)
> 
> and then use a dark mode mixin. What do you think?

I wouldn't say we should do it like this. Better would be to adjust the calculation to work correctly in all, dark and bright mode.

**Comment by susnux** [comment]
I agree with @szaimen as that is the purpose of the `--color-primary-element`. To be used for graphical elements and thus should have a high enough  contrast

**Comment by JuliaKirschenheuter** [comment]
checked several components on a dark theme, sometimes there are 3.02:1 but still passing criteria. Seems not to be an issue anymore.

**Comment by JuliaKirschenheuter** [comment]
seems to be not resolved on activity settings on the table

**Comment by JuliaKirschenheuter** [comment]
Dear @nimishavijay 

we have 2 problems in a dark theme in combination with `--color-primary-element` (`#00679e`):

`--color-background-hover` -> `#212121`

![Screenshot from 2023-12-15 15-56-16](https://github.com/nextcloud/server/assets/6078378/c1018f97-3347-43a5-a771-ab0ec81d8d94)

`--color-main-background` -> `#171717`

![Screenshot from 2023-12-15 15-57-09](https://github.com/nextcloud/server/assets/6078378/c8bc6fdc-e44c-4e90-a5c9-d43ad9a9bd85)


You can check it on hovered and normal state:

![Screenshot from 2023-12-15 17-14-41](https://github.com/nextcloud/server/assets/6078378/248ce42b-b877-4df6-a007-ff4e961f7ff8)
![Screenshot from 2023-12-15 17-14-23](https://github.com/nextcloud/server/assets/6078378/0101e99d-87da-43fc-ba08-322b7325a3e1)

Could you please provide another colors for us?
Probably related:

https://play.phpsandbox.io/mexitek/phpcolors
https://github.com/nextcloud/server/pull/39317
https://github.com/nextcloud/server/pull/39391

**Comment by susnux** [comment]
Might be fixed with https://github.com/nextcloud/server/pull/42287

**Comment by JuliaKirschenheuter** [comment]
closed https://github.com/nextcloud/server/pull/42287
